File: C++/Stack/Stack_01.cpp
////////////////////////////////////////////////////////////
//
//  Program Name    : Stack Using Singly Linear Linked List
//  Program Number  : 001
//  Description     : Implementation of Stack Data Structure
//                    using Singly Linear Linked List.
//
//                    Operations Performed:
//                    1. Push
//                    2. Pop
//                    3. Peep
//                    4. Display
//                    5. Count
//
//  Concepts Used   : Class, Dynamic Memory Allocation,
//                    Linked List, Stack (LIFO)
//
//  Input           : Integer Data
//  Output          : Stack Operations Result
//
//  Language        : C++
//  Compiler        : GNU G++
//
//  Author          : Atharv Tushar Bhosale
//  Date            : August 2026
//
////////////////////////////////////////////////////////////

// Stack 

#include<iostream>

using namespace std;

#pragma pack(1)
struct node
{
    int data;
    struct node* next;
};

class Stack
{
    private:
        struct node* first;
        int iCount;

    public:
        Stack();
        void Push(int iNo);             // Insert First
        int Pop();                     // Delete First
        int Peep();                    // Delete First (it will give data only not deleting data)
        void Display();
        int Count();
};

Stack :: Stack()
{}

void Stack :: Push(int iNo)           
{}

int Stack :: Pop()
{
    return 0;
}           

int Stack :: Peep()   
{}

void Stack :: Display()
{}

int Stack :: Count()
{
    return iCount;
}

int main()
{
    Stack sobj;
    return 0;
}

////////////////////////////////////////////////////////////
//
//  End of Program
//
//  Stack Using Singly Linear Linked List
//
//  Author : Atharv Tushar Bhosale
//  Date   : August 2026
//
////////////////////////////////////////////////////////////

File: Java/Stack/Stack_01.java
/*
===============================================================================
Program Name    : Stack Skeleton
Program Number  : 001
File Name       : Stack_01.java

Description     : This program demonstrates the basic
                  structure of Stack using Singly
                  Linear Linked List.

Concepts Used:
1. Class and Object
2. Linked List
3. Stack Data Structure

Date            : August 2026
Language        : Java
Author          : Atharv Tushar Bhosale
===============================================================================
*/

class node
{
    public int data;
    public node next;
}

class Stack
{
    private node first;
    private int iCount;

    public Stack()
    {
        first = null;
        iCount = 0;
    }
}

class Stack_01
{
    public static void main(String A[])
    {
        Stack sobj = new Stack();

        System.out.println("Stack created successfully");
    }
}

/*
===============================================================================
Sample Output

Stack created successfully

===============================================================================
End of Program
===============================================================================
*/
